.github/workflows/pages.yml

```diff
@@ -0,0 +1,35 @@
+name: Push to gh-pages
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get dependencies
+      run: |
+        cd ./emulator
+        curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+        wasm-pack build -t web
+        cd ../emulator-webapp
+        npm install --legacy-peer-deps
```

Comment:
The project uses the yarn package manager instead of npm. I'm not sure if this might cause an issue

Reply:
Sorry I was unaware of this and have now replaced the corresponding line. This also stopped the legacy warning to be an issue :)